src/events/provider.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Spark Service accounts related event handlers."""
+
+from ops import CharmBase
+
+from common.utils import WithLogging
+from constants import SERVICE_ACCOUNT
+from core.context import Context
+from core.workload import ConfigurationHubWorkloadBase
+from events.base import BaseEventHandler
+from managers.configuration_hub import ConfigurationHubManager
+from relations.spark_sa import (
+    ServiceAccountProvider,
+    ServiceAccountReleasedEvent,
+    ServiceAccountRequestedEvent,
+)
+
+
+class ServiceAccountsEvents(BaseEventHandler, WithLogging):
+    """Class implementing Spark Service Account Integration event hooks."""
+
+    def __init__(self, charm: CharmBase, context: Context, workload: ConfigurationHubWorkloadBase):
+        super().__init__(charm, "service-account")
+
+        self.charm = charm
+        self.context = context
+        self.workload = workload
+
+        self.configuration_hub = ConfigurationHubManager(self.workload)
```

Comment:
**question** This is weird, why is this needed? Do we need to restart the service or update configurations?